File: dashboard/src/api/tickets.ts
import { getJson } from './client'
import type { TicketDetailResponse, TicketListResponse } from './types'

export interface ListTicketsParams {
  customer_id?: string
  status?: string
  limit?: number
  offset?: number
}

export function listTickets(params: ListTicketsParams = {}, signal?: AbortSignal) {
  const search = new URLSearchParams()
  if (params.customer_id) search.set('customer_id', params.customer_id)
  if (params.status) search.set('status', params.status)
  search.set('limit', String(params.limit ?? 50))
  search.set('offset', String(params.offset ?? 0))
  return getJson<TicketListResponse>(`/api/tickets?${search.toString()}`, signal)
}

export function getTicket(ticketId: string, signal?: AbortSignal) {
  return getJson<TicketDetailResponse>(`/api/tickets/${ticketId}`, signal)
}

File: api/routers/tickets.py
from __future__ import annotations

import uuid

from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from api.deps import get_db
from api.schemas.interactions import TicketDetailResponse
from api.schemas.tickets import TicketListResponse
from api.services import tickets_service
from recommender.models import TicketStatus

router = APIRouter(prefix="/api/tickets", tags=["tickets"])


@router.get("", response_model=TicketListResponse)
def list_tickets(
    customer_id: uuid.UUID | None = None,
    status: TicketStatus | None = None,
    limit: int = Query(default=50, le=200),
    offset: int = Query(default=0, ge=0),
    session: Session = Depends(get_db),
) -> TicketListResponse:
    return tickets_service.list_tickets(session, customer_id, status, limit, offset)


@router.get("/{ticket_id}", response_model=TicketDetailResponse)
def get_ticket(ticket_id: uuid.UUID, session: Session = Depends(get_db)) -> TicketDetailResponse:
    return tickets_service.get_ticket(session, ticket_id)
